src/Mobs/PathFinder.cpp

```diff
@@ -0,0 +1,233 @@
+
+
+#include "Globals.h"
+#include "PathFinder.h"
+#include "../Chunk.h"
+
+cPathFinder::cPathFinder(double a_MobWidth, double a_MobHeight)
+	: m_Path(nullptr)
+	, m_GiveUpCounter(0)
+	, m_TicksSinceLastPathReset(1000)
```

Comment:
My suggestion remains the same, to recalculate if there is a change in where the entity should be going.

I suppose it comes down to a matter of performance versus accuracy. For selective recalculation, you'll definitely encounter terrain layouts where even a one block change in destination could mean a completely different path, and so a player would see a monster walking towards them, stop, and turn back to head along a different route. This may or may not be noticeable depending on how large X is set to be. For immediate recalculation, the monster may not move at all if the pathfinder is particularly slow and the player is in constant motion, but that really depends on exactly how fast cPath currently is, which I have no idea.

So in summary, I haven't enough data to really know. :/
Perhaps you could do some tests? :)